services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/extensions/dynamodb/mappingclient/operations/BatchWriteItemOperation.java

```diff
@@ -0,0 +1,98 @@
+import software.amazon.awssdk.extensions.dynamodb.mappingclient.MappedTableResource;
+import software.amazon.awssdk.extensions.dynamodb.mappingclient.MapperExtension;
+import software.amazon.awssdk.extensions.dynamodb.mappingclient.OperationContext;
+import software.amazon.awssdk.extensions.dynamodb.mappingclient.model.BatchWriteItemEnhancedRequest;
+import software.amazon.awssdk.extensions.dynamodb.mappingclient.model.BatchWriteResult;
+import software.amazon.awssdk.extensions.dynamodb.mappingclient.model.WriteBatch;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.BatchWriteItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.BatchWriteItemResponse;
+import software.amazon.awssdk.services.dynamodb.model.WriteRequest;
+
+@SdkInternalApi
+public class BatchWriteItemOperation implements DatabaseOperation<BatchWriteItemRequest,
+                                 BatchWriteItemResponse, BatchWriteResult> {
+
+    private final BatchWriteItemEnhancedRequest request;
+
+    private BatchWriteItemOperation(BatchWriteItemEnhancedRequest request) {
+        this.request = request;
+    }
+
+    public static BatchWriteItemOperation create(BatchWriteItemEnhancedRequest request) {
+        return new BatchWriteItemOperation(request);
+    }
+
+    @Override
+    public BatchWriteItemRequest generateRequest(MapperExtension mapperExtension) {
+        Map<String, Collection<WriteRequest>> requestItems = new HashMap<>();
+        request.writeBatches().forEach(writeBatch -> addWriteRequestsToMap(writeBatch, requestItems));
+        return BatchWriteItemRequest.builder()
+                                    .requestItems(Collections.unmodifiableMap(requestItems))
+                                    .build();
+    }
+
+    @Override
+    public BatchWriteResult transformResponse(BatchWriteItemResponse response, MapperExtension mapperExtension) {
+        return BatchWriteResult.builder().unprocessedRequests(response.unprocessedItems()).build();
+    }
+
+    @Override
+    public Function<BatchWriteItemRequest, BatchWriteItemResponse> serviceCall(DynamoDbClient dynamoDbClient) {
+        return dynamoDbClient::batchWriteItem;
+    }
+
+    @Override
+    public Function<BatchWriteItemRequest, CompletableFuture<BatchWriteItemResponse>> asyncServiceCall(
+        DynamoDbAsyncClient dynamoDbAsyncClient) {
+
+        return dynamoDbAsyncClient::batchWriteItem;
+    }
+
+    // Due to raw-type erasure, it's necessary to pass a map in to be mutated rather than try and extract the write
+    // requests which would be more straight forward but Collection<WriteRequest> becomes Collection when you try and
+    // exfiltrate the objects from a raw-typed WriteBatch.
+    private void addWriteRequestsToMap(WriteBatch writeBatch, Map<String, Collection<WriteRequest>> writeRequestMap) {
```

Comment:
The type erasure problem no longer exists due to this refactor, therefore we should be able to do a straight map into WriteRequests